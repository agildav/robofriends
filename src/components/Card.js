import React from "react";
import "tachyons";

const Card = ({ name, nickname }) => {
  return (
    <div className="bg-light-green dib br4 pa3 ma2 bw3 shadow-5 grow">
      <img
        //  Get image and set size, background
        src={`https://robohash.org/${name}*${nickname}?size=250x250&bgset=bg1`}
        alt="robot"
      />
      <div>
        <h2 className="b">{name}</h2>
        <p>{nickname}</p>
      </div>
    </div>
  );
};

export default Card;
