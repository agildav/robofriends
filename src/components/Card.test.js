import { shallow } from "enzyme";
import React from "react";
import Card from "./Card";

test("Card component is rendered", () => {
  expect.assertions(1);
  return expect(shallow(<Card />).length).toEqual(1);
});
