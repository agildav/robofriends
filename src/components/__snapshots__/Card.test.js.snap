// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component is rendered 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt="robot"
          src="https://robohash.org/undefined*undefined?size=250x250&bgset=bg1"
/>,
        <div>
          <h2
                    className="b"
          />
          <p />
</div>,
      ],
      "className": "bg-light-green dib br4 pa3 ma2 bw3 shadow-5 grow",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "robot",
          "src": "https://robohash.org/undefined*undefined?size=250x250&bgset=bg1",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2
              className="b"
/>,
            <p />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "b",
            },
            "ref": null,
            "rendered": null,
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "p",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="robot"
            src="https://robohash.org/undefined*undefined?size=250x250&bgset=bg1"
/>,
          <div>
            <h2
                        className="b"
            />
            <p />
</div>,
        ],
        "className": "bg-light-green dib br4 pa3 ma2 bw3 shadow-5 grow",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "robot",
            "src": "https://robohash.org/undefined*undefined?size=250x250&bgset=bg1",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2
                className="b"
/>,
              <p />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "b",
              },
              "ref": null,
              "rendered": null,
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "p",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
